### PR TITLE
LPD-37443 property for related element ERC is not documented in the schema

### PR DESCRIPTION
**Pull request by movaldivia**
https://liferay.atlassian.net/browse/LPD-37443

### CODE REVIEW

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
Congratulations on your first PR!

We do not do merge commits, please rebase on top of master and do `git push -f`.

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/openapi/v1_0/ObjectEntryOpenAPIResourceImpl.java [hunk 1]
```diff
@@ -362,6 +362,17 @@ private OpenAPISchemaFilter _getOpenAPISchemaFilter(
...
+					});
+
+				dtoProperties.add(
+					new DTOProperty(
+						Collections.singletonMap("x-parent-map", "properties"),
+						objectRelationship.getName() + "ERC",
+						String.class.getSimpleName()) {
+
 						{
 							setRequired(objectField.isRequired());
```
Comment (ccorreagg) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/openapi/v1_0/ObjectEntryOpenAPIResourceImpl.java:377]: I am not sure if the field is required... I think should be read only.

I have created two object definitions, Student and University and created a relationship (one to many) between them. Then, I have created a university and I got the ERC. Then, I try to create a Student and send this field with the ERC of the university:

```curl -X POST -u test@liferay.com:test http://localhost:8080/o/c/students -H 'Content-type: application/json' -d '{"studentName": "student6", "universityStudentsERC": "4cb30281-d2cf-b77f-3e95-e576b9d67803"}'```

Response:

```
{
 [...]
  "externalReferenceCode" : "f1adb466-3e83-7ab0-213b-7fd8759d5c3f",
  "id" : 33806,
  [...]
  "r_universityStudents_c_universityId" : 0,
  "universityStudentsERC" : "",
  "studentName" : "student6",
  "r_universityStudents_c_universityERC" : ""
}
```

The result is the student is NOT related to the university. If I use the `r_universityStudents_c_universityERC` field instead, it works. So I would say the best approach is to set this field as read only and not mandatory. What do you think?

cc/ @4lejandrito

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:sf

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:test:relevant

**Comment by movaldivia** [comment]
ci:stop

**Comment by movaldivia** [comment]
ci:test:stop

**Comment by movaldivia** [comment]
I think that the failed test are not related to this pr changes.

Here is another pr with similar errors: https://testray.liferay.com/#/project/35392/routines/45357/build/29015271

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
@movaldivia I think the semantic versioning failures are related. You have to run `gw baseline` in the `portal-vulcan-api` module. Baseline will update the versions based on the changes (major, minor or patch).

**Comment by movaldivia** [comment]
ci:test:relevant

**Review by ccorreagg — APPROVED** [review]

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
@movaldivia Remember also to launch the sf suite when you update the PR

**Comment by ccorreagg** [comment]
The failures seems they are not related. Forwarding the PR

**Comment by ccorreagg** [comment]
ci:forward:force

**Pull request closed without merging**